tyxml/snabbdom_tyxml.ml

```diff
@@ -0,0 +1,74 @@
+open Snabbdom
+
+module Xml_vnode = struct
+  module W = Xml_wrap.NoWrap
+
+  type 'a wrap = 'a
+  type 'a list_wrap = 'a list
+  type uri = string
+
+  let uri_of_string s = s
+  let string_of_uri s = s
+
+  type event_handler = Jv.t -> unit
+  type mouse_event_handler = Brr.Ev.Mouse.t Brr.Ev.t -> unit
+  type keyboard_event_handler = Brr.Ev.Keyboard.t Brr.Ev.t -> unit
+  type touch_event_handler = Jv.t -> unit
+  type attrib = [ `Event of string | `Attr of string ] * Jv.t
+
+  let float_attrib name f = `Attr name, Jv.of_float f
+  let int_attrib name f = `Attr name, Jv.of_int f
+  let string_attrib name f = `Attr name, Jv.of_string f
+  let space_sep_attrib name xs = `Attr name, Jv.of_string (String.concat " " xs)
+  let comma_sep_attrib name xs = `Attr name, Jv.of_string (String.concat "," xs)
+  let event_handler_attrib name handler = `Event name, Jv.repr handler
+  let mouse_event_handler_attrib name handler = `Event name, Jv.repr handler
+  let touch_event_handler_attrib name handler = `Event name, Jv.repr handler
+  let keyboard_event_handler_attrib name handler = `Event name, Jv.repr handler
+  let uri_attrib name value = `Attr name, Jv.of_string value
+  let uris_attrib name values = `Attr name, Jv.of_string (String.concat " " values)
+
+  type aname = string
+  type elt = Vnode.t
+  type ename = string
+
+  let make_attrs (a : attrib list option) =
+    match a with
+    | None -> Jv.null
+    | Some attrs ->
+      let events, attrs =
+        List.fold_left
+          (fun (events, attrs) t ->
+            match t with
+            | `Attr name, v -> events, (name, v) :: attrs
```

Comment:
If I understand snabbdom's API right, you might have some issues with attributes such as `data-foo` and others. You probably want to recognize those specially.

Also, you should be able to extend your attribute type and provide all the fancy new things provided by the library as "general" (no type constraints) attributes.

Reply:
That is true. I was just testing with some simple attributes and event handlers at the moment. snabbdom seems to support a variety of attributes so i'll need a nice way to work with those both in the core library and the tyxml layer (i don't want to have people worry about creating `Jv.obj`'s and deal with javascript arrays and strings. 

I'd also want to deal with event handling attributes a little better. snabbdom uses event names like`click` and tyxml generates `onclick` and so on? 

> provide all the fancy new things provided by the library as "general" (no type constraints) attributes.

This sounds reasonable.

Reply:
For events, I would move the name mangling in the `*_event_handler_attrib` functions themselves, the rest is fine. For other attribs, you can use `to_attrib` to locally (and carefully!) break the abstraction and make the attributes you want.